react-counter/src/components/Counter.js

```diff
@@ -0,0 +1,30 @@
+import React, {Component} from 'react';
+import Plus from './Plus';
+import Minus from './Minus';
+import Refresh from './Refresh';
+import Display from './Display';
+
+class Counter extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {count: 0};
+  }
+
+  render() {
+    let {count} = this.state;
+    return (
+      <div className='main'>
+        <div className='display-wrapper'>
+          <Display count={count}/>
+        </div>
+        <div className='buttons-wrapper'>
+          <Plus onCountUp={() => {this.setState({ count: count + 1 })}}/>

```

Comment:
We shouldn't base on `this.state.count` , it could be changed in another time due to react optimizations. Instead, you should use a function: `this.setState((state) => ({ count: state.count + 1 }))`